SPECS/knem/knem.spec

```diff
@@ -0,0 +1,288 @@
+
+%install
+export INSTALL_MOD_PATH=$RPM_BUILD_ROOT
+export INSTALL_MOD_DIR=%install_mod_dir
+export KPNAME=%{_name}
+mkdir -p $RPM_BUILD_ROOT/etc/udev/rules.d
+install -d $RPM_BUILD_ROOT/usr/lib64/pkgconfig
+for flavor in %flavors_to_build; do
+	cd $PWD/obj/$flavor
+	export KSRC=%{kernel_source $flavor}
+	export KVERSION=%{kernel_release $KSRC}
+	make DESTDIR=$RPM_BUILD_ROOT install KERNELRELEASE=$KVERSION
+	export MODULE_DESTDIR=/lib/modules/$KVERSION/$INSTALL_MOD_DIR
+	mkdir -p $RPM_BUILD_ROOT/lib/modules/$KVERSION/$INSTALL_MOD_DIR
+	MODULE_DESTDIR=/lib/modules/$KVERSION/$INSTALL_MOD_DIR DESTDIR=$RPM_BUILD_ROOT KVERSION=$KVERSION $RPM_BUILD_ROOT/opt/knem-%{version}/sbin/knem_local_install
+	cp knem.pc  $RPM_BUILD_ROOT/usr/lib64/pkgconfig
+	cd -
+done
+
+/bin/rm -rf %{buildroot}/opt/knem-%{version}/lib/modules || true
+
+%if %{IS_RHEL_VENDOR}
+# Set the module(s) to be executable, so that they will be stripped when packaged.
+find %{buildroot} \( -type f -name '*.ko' -o -name '*ko.gz' \) -exec %{__chmod} u+x \{\} \;
+
+%if ! 0%{?fedora}
+%{__install} -d $RPM_BUILD_ROOT%{_sysconfdir}/depmod.d/
+echo "override knem * weak-updates/%{_name}" >> $RPM_BUILD_ROOT%{_sysconfdir}/depmod.d/%{_name}.conf
+echo "override knem * extra/%{_name}" >> $RPM_BUILD_ROOT%{_sysconfdir}/depmod.d/%{_name}.conf
+%endif
+%else
+find %{buildroot} \( -type f -name '*.ko' -o -name '*ko.gz' \) -exec %{__strip} -p --strip-debug --discard-locals -R .comment -R .note \{\} \;
+%endif
+
+%clean
+rm -rf $RPM_BUILD_ROOT
+
+%post
+getent group rdma >/dev/null 2>&1 || groupadd -r rdma
+touch /etc/udev/rules.d/10-knem.rules
+# load knem
+/sbin/modprobe -r knem > /dev/null 2>&1
+/sbin/modprobe knem > /dev/null 2>&1
+
+# automatically load knem onboot
+if [ -d /etc/sysconfig/modules ]; then
+	# RH
+	echo "/sbin/modprobe knem > /dev/null 2>&1" > /etc/sysconfig/modules/knem.modules
+	chmod +x /etc/sysconfig/modules/knem.modules
+elif [ -e /etc/sysconfig/kernel ]; then
+	# SLES
+	if ! (grep -w knem /etc/sysconfig/kernel); then
+		sed -i -r -e 's/^(MODULES_LOADED_ON_BOOT=)"(.*)"/\1"\2 knem"/' /etc/sysconfig/kernel
+	fi
+fi
+
+%preun
+# unload knem
+/sbin/modprobe -r knem > /dev/null 2>&1
+# RH
+/bin/rm -f /etc/sysconfig/modules/knem.modules
+# SLES
+if (grep -qw knem /etc/sysconfig/kernel 2>/dev/null); then
+	sed -i -e 's/ knem//g' /etc/sysconfig/kernel 2>/dev/null
+fi
+
+%if "%{KMP}" != "1"
+%post -n %{non_kmp_pname}
+depmod %{KVERSION} -a
+
+%postun -n %{non_kmp_pname}
+if [ $1 = 0 ]; then  # 1 : Erase, not upgrade
+	depmod %{KVERSION} -a
+fi
+%endif # end KMP=1
+
+%files
+%defattr(-, root, root)
+%license source/COPYING source/COPYING.BSD-3 source/COPYING.GPL-2
+/opt/knem-%{version}
+/usr/lib64/pkgconfig/knem.pc
+
+%config(noreplace)
+/etc/udev/rules.d/10-knem.rules
+
+
+%if "%{KMP}" != "1"
+%files -n %{non_kmp_pname}
+/lib/modules/%{KVERSION}/%{install_mod_dir}/
+%if %{IS_RHEL_VENDOR}
+%if ! 0%{?fedora}
+%config(noreplace) %{_sysconfdir}/depmod.d/%{_name}.conf
+%endif
+%endif
+%endif
```

Comment:
Missing `%license`. Not strictly necessary when there's a signed spec counterpart to this one but consistency helps.

Reply:
Added license file.